File: MyLMS/Scripts/AngularControllers/OrganisationController.js
myapp.controller('OrganisationCntrl', function ($scope, $http) {
    GetEntityList();
    $scope.SaveEntity = function () {
        debugger;
        var _EntityName = $scope.EntityName
        var _EntityCode = $scope.EntityCode;
        var _ManagerName = $scope.ManagerName;
        var _Email = $scope.Email;
        var _Mobile = $scope.Mobile;
        var _Landline = $scope.Landline;
        var _Address = $scope.Address;
        
        $http({
            method: 'POST',
            url: '/Organisation/CreateEntity',
            data: { EntityName: _EntityName, EntityCode: _EntityCode, ManagerName: _ManagerName, Email: _Email, Mobile: _Mobile, Landline: _Landline, Address: _Address }
        }).then(function (result) {
            GetEntityList();
        });
    }

    function GetEntityList() {
        $http.get('/Organisation/GetEntityList')
        .then(function (result) {
            $scope.EntityList = result.data;
        });
    }

    $scope.OpenViewEntityList = function () {
        window.location.href = "/Organisation/Entity/";
    }

    $scope.GetEntityDetails = function () {
        var _UserID = $scope.EntityID;
        for (var i = 0; i < $scope.EntityList.length; i++) {
            e = $scope.EntityList[i];
            console.log(e.EntityID);
            if (e.EntityID == _UserID) {
                $scope.EntityName = e.EntityName;
                $scope.EntityCode = e.EntityCode;
                $scope.ManagerName = e.ManagerName;
                $scope.Email = e.Email;
                $scope.Mobile = e.Mobile;
                $scope.Landline = e.Landline;
                $scope.Address = e.Address;
            }
        }
    }

    $scope.EditEntity = function () {
        var _EntityID = $scope.EntityID;
        var _EntityName = $scope.Email;
        var _EntityCode = $scope.EntityCode;
        var _ManagerName = $scope.ManagerName;
        var _Email = $scope.Email;
        var _Mobile = $scope.Mobile;
        var _Landline = $scope.Landline;
        var _Address = $scope.Address;

        $http({
            method: 'POST',
            url: '/Organisation/EditEntity',
            data: { EntityID: _EntityID, EntityName: _EntityName, EntityCode: _EntityCode, ManagerName: _ManagerName, Email: _Email, Mobile: _Mobile, Landline: _Landline, Address: _Address }
        }).then(function (result) {
            alert("Updated!");
            GetEntityList();
        });
    }
});

myapp.controller('OrgUserCntrl', function ($scope, $http) {
    GetOrgList();
    $scope.EntityTextToShow = "Please wait.."
    $scope.RoleTextToShow = "Please wait.."


    function GetOrgList() {
        $http.get('/Organisation/GetOrgUserList')
            .then(function (result) {
                $scope.OrgUserList = result.data;
            });
    }

    $scope.AddOrgUser = function () {
        debugger;

        var _UserName = $scope.EmailID;
        var _Password = $scope.Password;
        var _FullName = $scope.FullName;
        var _EmailID = $scope.EmailID;
        var _Mobile = $scope.Mobile;

        $http({
            method: 'POST',
            url: '/Organisation/AddOrgUser',
            data: { UserName: _UserName, Password: _Password, FullName: _FullName, EmailID: _EmailID, Mobile: _Mobile }
        }).then(function (result) {
            $http.get('/Organisation/GetOrgUserList')
                .then(function (result) {
                    $scope.OrgUserList = result.data;
                });
        });
    }

    $scope.GetOrgUserDetails = function () {
        var _UserID = $scope.UserID;
        for (var i = 0; i < $scope.OrgUserList.length; i++) {
            e = $scope.OrgUserList[i];
            console.log(e.UserID);
            if (e.UserID == _UserID) {
                $scope.FullName = e.FullName;
                $scope.UserName = e.UserName;
                $scope.Password = e.Password;
                $scope.EmailID = e.EmailID;
                $scope.Mobile = e.Mobile;
                $scope.RoleID = e.RoleID;
            }
        }
    }

    $scope.EditOrgUser = function () {
        var _UserName = $scope.EmailID;
        var _Password = $scope.Password;
        var _FullName = $scope.FullName;
        var _EmailID = $scope.EmailID;
        var _Mobile = $scope.Mobile;
        var _RoleID = $scope.RoleID;
        var _UserID = $scope.UserID;

        $http({
            method: 'POST',
            url: '/Organisation/EditUser',
            data: { UserName: _UserName, Password: _Password, FullName: _FullName, EmailID: _EmailID, Mobile: _Mobile, RoleID: _RoleID, UserID: _UserID }
        }).then(function (result) {
            alert("Updated!")
            $http.get('/Organisation/GetOrgUserList')
                .then(function (result) {
                    $scope.OrgUserList = result.data;
                });
        });

    }
    $scope.DeleteUser = function (UserID) {
        if (confirm("Are you sure you want to Delete this User?"))
            $http({
                method: 'POST',
                url: '/Organisation/DeleteUser',
                data: { DeleteUserID: UserID }
            }).then(function (result) {
                alert('Deleted Successfully!!');
                GetOrgList();
            });
    }

});

myapp.controller('EntityUserCntrl', function ($scope, $http) {
    GetEntityList();
    $scope.EntityTextToShow = "Please wait.."
    $scope.RoleTextToShow = "Please wait.."


    function GetEntityList() {
        $http.get('/Organisation/GetEntityList')
        .then(function (result) {
            $scope.EntityList = result.data;
            $scope.EntityTextToShow = "Please select entity.."
        });
    }

    $scope.GetEntityUsers = function GetEntityUsersList() {
        $http.get('/Organisation/GetEntityUserList/' + $scope.EntityID)
        .then(function (result) {
            $scope.EntityUserList = result.data;
            GetRoles();
        });
    }

    $scope.GetEntityUserDetails = function () {
        var _UserID = $scope.UserID;
        for (var i = 0; i < $scope.EntityUserList.length; i++) {
            e = $scope.EntityUserList[i];
            console.log(e.UserID);
            if (e.UserID == _UserID) {
                $scope.FullName = e.FullName;
                $scope.UserName = e.UserName;
                $scope.Password = e.Password;
                $scope.EmailID = e.EmailID;
                $scope.Mobile = e.Mobile;
            }
        }

    }

    function GetRoles() {
        $http.get('/Organisation/GetEntityAdminRole')
        .then(function (result) {
            $scope.EntityAdminUsers = result.data;
            $scope.RoleTextToShow = "Please select.."
        });
    }

    $scope.AddEntityUser = function () {
        debugger;
        var _EntityID = $scope.EntityID;
        var _UserName = $scope.EmailID;
        var _Password = $scope.Password;
        var _FullName = $scope.FullName;
        var _EmailID = $scope.EmailID;
        var _Mobile = $scope.Mobile;
        var _RoleID = $scope.RoleID;

        $http({
            method: 'POST',
            url: '/Organisation/AddEntityUser',
            data: {EntityID: _EntityID, UserName: _UserName, Password: _Password, FullName: _FullName, EmailID: _EmailID, Mobile: _Mobile, RoleID: _RoleID }
        }).then(function (result) {
            alert("User saved successfully!");
            $http.get('/Organisation/GetEntityUserList/' + $scope.EntityID)
            .then(function (result) {
                $scope.EntityUserList = result.data;
                GetRoles();
            });
        });
    }

    $scope.EditEntityUser = function () {
        debugger;
        var _UserName = $scope.EmailID;
        var _Password = $scope.Password;
        var _FullName = $scope.FullName;
        var _EmailID = $scope.EmailID;
        var _Mobile = $scope.Mobile;
        var _RoleID = $scope.RoleID;
        var _UserID = $scope.UserID;

        $http({
            method: 'POST',
            url: '/Organisation/EditUser',
            data: { UserName: _UserName, Password: _Password, FullName: _FullName, EmailID: _EmailID, Mobile: _Mobile, RoleID: _RoleID, UserID: _UserID }
        }).then(function (result) {
            alert("Updated!")
            $http.get('/Organisation/GetEntityUserList/' + $scope.EntityID)
                .then(function (result) {
                    $scope.EntityUserList = result.data;
                    GetRoles();
                });
        });
    }
    $scope.DeleteUser = function (UserID) {
        if (confirm("Are you sure you want to Delete this User?"))
            $http({
                method: 'POST',
                url: '/Organisation/DeleteUser',
                data: { DeleteUserID: UserID }
            }).then(function (result) {
                alert('Deleted Successfully!!');
                $scope.GetEntityUsers()
            });
    }

});

myapp.controller('ReceiverCntrl', function ($scope, $http) {
    GetEntityList();

    function GetEntityList() {
        $http.get('/Organisation/GetEntityList')
        .then(function (result) {
            $scope.EntityList = result.data;
            $scope.EntityTextToShow = "Please select entity.."
        });
    }

    $scope.GetReceiversList = function () {
        $http.get('/Organisation/GetReceiversList/' + $scope.EntityID)
        .then(function (result) {
            $scope.ReceiversList = result.data;
        });
    }

    $scope.SaveReceiver = function () {
        debugger;
        var _ReceiverSerialNo = $scope.ReceiverSerialNo
        var _ReceiverModel = $scope.ReceiverModel
        var _EntityID = $scope.SelectedEntityID

        $http({
            method: 'POST',
            url: '/Organisation/SaveShippedReceiver',
            data: { ReceiverSerialNo: _ReceiverSerialNo, ReceiverModel: _ReceiverModel, EntityID: _EntityID }
        }).then(function (result) {
            $http.get('/Organisation/GetReceiversList/' + $scope.EntityID)
               .then(function (result) {
                   $scope.ReceiversList = result.data;
               });
        });
    }
});

myapp.controller('RemoteCntrl', function ($scope, $http) {
    GetEntityList();

    function GetEntityList() {
        $http.get('/Organisation/GetEntityList')
        .then(function (result) {
            $scope.EntityList = result.data;
            $scope.EntityTextToShow = "Please select entity.."
        });
    }

    $scope.GetRemotesList = function () {
        $http.get('/Organisation/GetRemotesList/' + $scope.EntityID)
        .then(function (result) {
            $scope.RemotesList = result.data;
        });
    }

    $scope.SaveRemote = function () {
        debugger;
        var _ShipmentAmount = $scope.ShipmentAmount
        var _RemoteModel = $scope.RemoteModel
        var _EntityID = $scope.SelectedEntityID

        $http({
            method: 'POST',
            url: '/Organisation/SaveShippedRemote',
            data: { ShipmentAmount: _ShipmentAmount, RemoteModel: _RemoteModel, EntityID: _EntityID }
        }).then(function (result) {
            $http.get('/Organisation/GetRemotesList/' + $scope.EntityID)
               .then(function (result) {
                   $scope.RemotesList = result.data;
               });
        });
    }
});

File: MyLMS/Controllers/OrganisationController.cs
using MyLMS.Models;
using Newtonsoft.Json;
using System;
using System.Data;
using System.Data.SqlClient;
using System.Web.Mvc;
using UtilityClass;

namespace MyLMS.Controllers
{
    public class OrganisationController : Controller
    {
        // GET: Organisation
        public ActionResult Index()
        {
            return View();
        }

        public ActionResult Entity()
        {
            return View();
        }

        public ActionResult CreateEntityUser()
        {
            return View();
        }

        public ActionResult ShippingReceiver()
        {
            return View();
        }

        public ActionResult ShippingRemote()
        {
            return View();
        }
        public ActionResult CreateOrgUser()
        {
            return View();
        }

        [HttpPost]
        public void CreateEntity(string EntityName, string EntityCode, string ManagerName, string Email, string Mobile, string Landline, string Address)
        {
            OrganisationModel OrgObj = new OrganisationModel();            
            SqlParameter[] SParam = new SqlParameter[8];

            SParam[0] = new SqlParameter("@UserID", SqlDbType.Int);
            SParam[0].Value = Convert.ToInt32(Session["USER_ID"]);
            SParam[1] = new SqlParameter("@EntityName", SqlDbType.VarChar);
            SParam[1].Value = EntityName;
            SParam[2] = new SqlParameter("@EntityCode", SqlDbType.VarChar);
            SParam[2].Value = EntityCode;
            SParam[3] = new SqlParameter("@ManagerName", SqlDbType.VarChar);
            SParam[3].Value = ManagerName;
            SParam[4] = new SqlParameter("@Email", SqlDbType.VarChar);
            SParam[4].Value = Email;
            SParam[5] = new SqlParameter("@Mobile", SqlDbType.VarChar);
            SParam[5].Value = Mobile;
            SParam[6] = new SqlParameter("@Landline", SqlDbType.VarChar);
            SParam[6].Value = Landline;
            SParam[7] = new SqlParameter("@Address", SqlDbType.Text);
            SParam[7].Value = Address;

            try
            {
                OrgObj.CreateEntity(SParam);
            }
            catch (Exception ex)
            {

            }
        }

        [HttpGet]
        public string GetOrgUserList()
        {
            DataTable OrgUserList = DAL.GetDataTable("GetOrgUserList");

            string JSONString = string.Empty;
            JSONString = JsonConvert.SerializeObject(OrgUserList);
            return JSONString;
        }

        [HttpGet]
        public string GetEntityList()
        {
            SqlParameter[] FObj = new SqlParameter[1];
            FObj[0] = new SqlParameter("@UserID", SqlDbType.Int);
            FObj[0].Value = Convert.ToInt32(Session["USER_ID"]);
            DataTable EntityList = DAL.GetDataTable("GetEntityList", FObj);

            string JSONString = string.Empty;
            JSONString = JsonConvert.SerializeObject(EntityList);
            return JSONString;
        }

        [HttpGet]
        public string GetEntityUserList(int ID)
        {
            SqlParameter[] FObj = new SqlParameter[1];
            FObj[0] = new SqlParameter("@EntityID", SqlDbType.Int);
            FObj[0].Value = ID;
            DataTable EntityUserList = DAL.GetDataTable("GetEntityUserList", FObj);

            string JSONString = string.Empty;
            JSONString = JsonConvert.SerializeObject(EntityUserList);
            return JSONString;
        }

        [HttpGet]
        public string GetEntityAdminRole()
        {
            DataTable EntityUserList = DAL.GetDataTable("GetEntityAdminRole");

            string JSONString = string.Empty;
            JSONString = JsonConvert.SerializeObject(EntityUserList);
            return JSONString;
        }

        [HttpPost]
        public void AddOrgUser(string UserName, string Password, string FullName, string EmailID, string Mobile)
        {
            OrganisationModel OrgObj = new OrganisationModel();
            SqlParameter[] SParam = new SqlParameter[6];

            SParam[0] = new SqlParameter("@UserID", SqlDbType.Int);
            SParam[0].Value = Convert.ToInt32(Session["USER_ID"]);
            SParam[1] = new SqlParameter("@UserName", SqlDbType.VarChar);
            SParam[1].Value = UserName;
            SParam[2] = new SqlParameter("@Password", SqlDbType.VarChar);
            SParam[2].Value = EDHelper.EncryptTripleDES(Password);
            SParam[3] = new SqlParameter("@FullName", SqlDbType.VarChar);
            SParam[3].Value = FullName;
            SParam[4] = new SqlParameter("@EmailID", SqlDbType.VarChar);
            SParam[4].Value = EmailID;
            SParam[5] = new SqlParameter("@Mobile", SqlDbType.VarChar);
            SParam[5].Value = Mobile;

            try
            {
                OrgObj.AddOrgUser(SParam);
            }
            catch (Exception ex)
            {

            }
        }

        [HttpPost]
        public void UpdatePassword(string UserID, string Password)
        {
            OrganisationModel OrgObj = new OrganisationModel();
            SqlParameter[] SParam = new SqlParameter[2];

            SParam[0] = new SqlParameter("@UserID", SqlDbType.Int);
            SParam[0].Value = UserID;
            SParam[1] = new SqlParameter("@Password", SqlDbType.VarChar);
            SParam[1].Value = EDHelper.EncryptTripleDES(Password);

            try
            {
                OrgObj.UpdatePassword(SParam);
            }
            catch (Exception ex)
            {

            }
        }


        [HttpPost]
        public void AddEntityUser(int EntityID, string UserName, string Password, string FullName, string EmailID, string Mobile, int RoleID)
        {
            OrganisationModel OrgObj = new OrganisationModel();
            SqlParameter[] SParam = new SqlParameter[8];

            SParam[0] = new SqlParameter("@UserID", SqlDbType.Int);
            SParam[0].Value = Convert.ToInt32(Session["USER_ID"]);
            SParam[1] = new SqlParameter("@UserName", SqlDbType.VarChar);
            SParam[1].Value = UserName;
            SParam[2] = new SqlParameter("@Password", SqlDbType.VarChar);
            SParam[2].Value = EDHelper.EncryptTripleDES(Password); 
            SParam[3] = new SqlParameter("@FullName", SqlDbType.VarChar);
            SParam[3].Value = FullName;
            SParam[4] = new SqlParameter("@EmailID", SqlDbType.VarChar);
            SParam[4].Value = EmailID;
            SParam[5] = new SqlParameter("@Mobile", SqlDbType.VarChar);
            SParam[5].Value = Mobile;
            SParam[6] = new SqlParameter("@RoleID", SqlDbType.VarChar);
            SParam[6].Value = RoleID;
            SParam[7] = new SqlParameter("@EntityID", SqlDbType.VarChar);
            SParam[7].Value = EntityID;

            try
            {
                OrgObj.AddEntityUser(SParam);
            }
            catch (Exception ex)
            {

            }
        }

        [HttpPost]
        public void EditUser(string UserName, string Password, string FullName, string EmailID, string Mobile, int RoleID, int UserID)
        {
            OrganisationModel OrgObj = new OrganisationModel();
            SqlParameter[] SParam = new SqlParameter[7];

            SParam[0] = new SqlParameter("@UserID", SqlDbType.Int);
            SParam[0].Value = UserID;
            SParam[1] = new SqlParameter("@UserName", SqlDbType.VarChar);
            SParam[1].Value = UserName;
            SParam[2] = new SqlParameter("@Password", SqlDbType.VarChar);
            SParam[2].Value = EDHelper.EncryptTripleDES(Password);
            SParam[3] = new SqlParameter("@FullName", SqlDbType.VarChar);
            SParam[3].Value = FullName;
            SParam[4] = new SqlParameter("@EmailID", SqlDbType.VarChar);
            SParam[4].Value = EmailID;
            SParam[5] = new SqlParameter("@Mobile", SqlDbType.VarChar);
            SParam[5].Value = Mobile;
            SParam[6] = new SqlParameter("@RoleID", SqlDbType.Int);
            SParam[6].Value = RoleID;

            try
            {
                OrgObj.EditUser(SParam);
            }
            catch (Exception ex)
            {

            }
        }

        [HttpGet]
        public string GetReceiversList(int ID)
        {
            SqlParameter[] FObj = new SqlParameter[2];

            FObj[0] = new SqlParameter("@UserID", SqlDbType.Int);
            FObj[0].Value = Convert.ToInt32(Session["USER_ID"]);
            FObj[1] = new SqlParameter("@EntityID", SqlDbType.Int);
            FObj[1].Value = ID;
            DataTable ReceiversList = DAL.GetDataTable("GetReceiverList", FObj);

            string JSONString = string.Empty;
            JSONString = JsonConvert.SerializeObject(ReceiversList);
            return JSONString;
        }

        [HttpGet]
        public string GetRemotesList(int ID)
        {
            SqlParameter[] FObj = new SqlParameter[2];

            FObj[0] = new SqlParameter("@UserID", SqlDbType.Int);
            FObj[0].Value = Convert.ToInt32(Session["USER_ID"]);
            FObj[1] = new SqlParameter("@EntityID", SqlDbType.Int);
            FObj[1].Value = ID;
            DataTable RemotesList = DAL.GetDataTable("GetRemoteList", FObj);

            string JSONString = string.Empty;
            JSONString = JsonConvert.SerializeObject(RemotesList);
            return JSONString;
        }

        [HttpPost]
        public void SaveShippedReceiver(string ReceiverSerialNo, string ReceiverModel, int EntityID)
        {
            OrganisationModel OrgObj = new OrganisationModel();
            SqlParameter[] SParam = new SqlParameter[4];

            SParam[0] = new SqlParameter("@UserID", SqlDbType.Int);
            SParam[0].Value = Convert.ToInt32(Session["USER_ID"]);
            SParam[1] = new SqlParameter("@ReceiverSerialNo", SqlDbType.VarChar);
            SParam[1].Value = ReceiverSerialNo;
            SParam[2] = new SqlParameter("@ReceiverModel", SqlDbType.VarChar);
            SParam[2].Value = ReceiverModel;
            SParam[3] = new SqlParameter("@EntityID", SqlDbType.VarChar);
            SParam[3].Value = EntityID;
           
            try
            {
                OrgObj.SaveShippedReceiver(SParam);
            }
            catch (Exception ex)
            {

            }
        }

        [HttpPost]
        public void SaveShippedRemote(int ShipmentAmount, string RemoteModel, int EntityID)
        {
            OrganisationModel OrgObj = new OrganisationModel();
            SqlParameter[] SParam = new SqlParameter[4];

            SParam[0] = new SqlParameter("@UserID", SqlDbType.Int);
            SParam[0].Value = Convert.ToInt32(Session["USER_ID"]);
            SParam[1] = new SqlParameter("@ShipmentAmount", SqlDbType.VarChar);
            SParam[1].Value = ShipmentAmount;
            SParam[2] = new SqlParameter("@RemoteModel", SqlDbType.VarChar);
            SParam[2].Value = RemoteModel;
            SParam[3] = new SqlParameter("@EntityID", SqlDbType.VarChar);
            SParam[3].Value = EntityID;

            try
            {
                OrgObj.SaveShippedRemote(SParam);
            }
            catch (Exception ex)
            {

            }
        }
    }
}